holoviews/core/spaces.py

```diff
@@ -513,6 +515,25 @@ def __call__(self, *args, **kwargs):
         return ret
 
 
+
+class Generator(Callable):
+    """
+    Generators are considered a special case of Callable that accept no
+    arguments and never memoize.
+    """
+
+    callable = param.Parameter(default=None, constant=True, doc="""
+         The generator function being wrapped.""")
+
+    @property
+    def argspec(self):
+        from inspect import ArgSpec
+        return ArgSpec(args=[], varargs=None, keywords=None, defaults=None)
+
+    def __call__(self):
+        return next(self.callable)
```

Comment:
Note that:

1. There is an *empty* stream expected on the DynamicMap but no *stream parameters*.
2. The user is responsible for initializing the generator with appropriate state.